Assets/HoloToolkit/Interaction/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,342 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+using System.Collections.Generic;
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity.Interaction;
+
+namespace HoloToolkit.Unity.Receivers
+{
+	/// <summary>
+	/// An interaction receiver is simply a component that attached to a list of interactible objects and does something
+	/// based on events from those interactible objects.  This is the base abstact class to extend from.
+	/// </summary>
+	public abstract class InteractionReceiver : MonoBehaviour
+	{
+		#region Public Members
+		/// <summary>
+		/// List of linked interactible objects to receive events for
+		/// </summary>
+		[Tooltip("Target Interactible Object to receive events for")]
+		public List<GameObject> Interactibles = new List<GameObject>();
+
+		/// <summary>
+		/// List of linked targets that the receiver affects
+		/// </summary>
+		[Tooltip("Targets for the receiver to ")]
+		public List<GameObject> Targets = new List<GameObject>();
+
+		/// <summary>
+		/// Flag for locking focus while selected
+		/// </summary>
+		[Tooltip("If true, this object will remain the prime focus while select is held")]
+		public bool bLockFocus;
+		#endregion
+
+		#region Private and Protected Members
+		/// <summary>
+		/// Internal protected member for our default gizmo icon
+		/// </summary>
+		protected string _gizmoIconDefault = "HUX/hux_receiver_icon.png";
+
+		/// <summary>
+		/// Internal protected member for our gizmo selected icon
+		/// </summary>
+		protected string _gizmoIconSelected = "HUX/hux_receiver_icon_selected.png";
+
+		/// <summary>
+		/// Protected string for the current active gizmo icon
+		/// </summary>
+		protected string _gizmoIcon;
+
+		/// <summary>
+		/// Protected focuser for the current selecting focuser
+		/// </summary>
+		protected IPointingSource _selectingFocuser;
+		#endregion
+
+		/// <summary>
+		/// On enable subscrible to all interaction events on elements in the interactibles list.
+		/// </summary>
+		public virtual void OnEnable()
+		{
+            InputManager.Instance.AddGlobalListener(gameObject);
```

Comment:
formatting.

Reply:
This should have been resolved with the last commit.  Should have proper spaces.